internal/controller/nodestate_cleaner_controller.go

```diff
@@ -96,3 +98,19 @@ func (r *NodeStateCleaner) SetupWithManager(mgr ctrl.Manager) error {
 		}).
 		Complete(r)
 }
+
+type PodSelector struct {
```

Comment:
Do we need really a new type? 
At the end can we have instead of 
```
	if err := r.List(ctx, pods, client.InNamespace(r.Namespace), client.MatchingLabels(map[string]string{
		r.FRRK8sSelector.Key: r.FRRK8sSelector.Value,
	})); err != nil {
```
something simpler like
```
	if err := r.List(ctx, pods, client.InNamespace(r.Namespace), frrk8sLabels); err != nil {
```
and avoid if possible custom parsing?

(some (AI) search shows https://pkg.go.dev/k8s.io/apimachinery/pkg/labels this could help)